students/Steven/lesson09/donor_models.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+from operator import itemgetter
+
+class Donor(object):
+    def __init__(self, donor_name, new_donation=None):
+        self.name = donor_name
+        if new_donation is None:
+            self.donation = []
+        else:
+            self.donation = list(new_donation)
+
+    def new_donation(self, new_amount):
+        update_donation = self.donation.append(new_amount)
+        return update_donation
+
+    @property
+    def donation_total(self):
+        return sum(self.donation)
+
+    @property
+    def donation_count(self):
+        return len(self.donation)
+
+    @property
+    def donation_avg(self):
+        return round(float(sum(self.donation) / self.donation_count), 2)
+
+    def thank_you_letter(self, donor_name, gift):
+        # Thank you email to donor for their gift
+        email = f"Thank you {donor_name} for your generous donation of ${gift:.2f}, your kindness is very appreciated." + "\n"
```

Comment:
you have the instance attribute for the name `self.name` - this function should not have `donor_name` input

Reply:
you can also grab the last element in self.donation for the gift value